apm-agent-plugins/apm-jdbc-plugin/src/main/java/co/elastic/apm/agent/jdbc/JdbcInstrumentation.java

```diff
@@ -26,18 +26,37 @@
 
 import co.elastic.apm.agent.bci.TracerAwareInstrumentation;
 import co.elastic.apm.agent.jdbc.helper.JdbcHelper;
+import net.bytebuddy.matcher.ElementMatcher;
 
+import javax.annotation.Nullable;
 import java.util.Collection;
 import java.util.Collections;
 
+import static co.elastic.apm.agent.bci.bytebuddy.CustomElementMatchers.classLoaderCanLoadClass;
+
 public abstract class JdbcInstrumentation extends TracerAwareInstrumentation {
 
     private static final Collection<String> JDBC_GROUPS = Collections.singleton("jdbc");
 
-    protected static JdbcHelper jdbcHelper = new JdbcHelper();
+    @Nullable
+    private static JdbcHelper jdbcHelper;
+
+    @Override
+    public ElementMatcher.Junction<ClassLoader> getClassLoaderMatcher() {
+        return classLoaderCanLoadClass("java.sql.Statement"); // in case java.sql module is not there
+    }
```

Comment:
Does this contradict the actual fix of the lazy loading? 
Meaning - the lazy loading assumes that `java.sql` classes should be available to JDBC code when it is invoked and the problem is that they are not yet in the classpath when the instrumented class is loaded. If that's the case, the tested class loader will be excluded, even though it shouldn't be,

Reply:
That's a fair point, I added this one when I expected there was an issue with java modules, but those weren't in use.

The issue was that the static call to `new JdbcHelper()` triggered loading JDBC classes in the wrong classloader, which I wasn't able to reproduce, even when using the same JVM.

Reply:
This extra check has been removed, and the fix is still working. Thus I've removed it in a later commit.